model_compression_toolkit/qat/keras/quantizer/ste_rounding/symmetirc_ste.py

```diff
@@ -143,3 +144,97 @@ def __call__(self,
+        self.threshold_values = quantization_config.activation_quantization_params[C.THRESHOLD]
+        self.threshold_shape = np.asarray(self.threshold_values).shape
+        self.np_threshold_values = float(self.threshold_values)
+
+        if self.power_of_two:
+            self.np_threshold_values = np.power(2.0,
+                                                np.ceil(np.log2(np.maximum(self.np_threshold_values, C.MIN_THRESHOLD))))
+        num_bits = quantization_config.activation_n_bits
+        delta = self.np_threshold_values / np.power(2.0, num_bits - int(C.WEIGHTS_SIGNED))
+        min_int = -int(C.WEIGHTS_SIGNED) * (2 ** (num_bits - int(C.WEIGHTS_SIGNED)))
+        max_int = (2 ** (num_bits - int(C.WEIGHTS_SIGNED))) - 1
+        self.min = delta * min_int
+        self.max = delta * max_int
+        self.quantizer_parameters = {}
+
+    def initialize_quantization(self,
+                                tensor_shape: TensorShape,
+                                name: str,
+                                layer: QuantizeWrapper) -> Dict[str, tf.Variable]:
+        """
+        Add min and max variables to layer.
+        Args:
+            tensor_shape: Tensor shape the quantizer quantize.
+            name: Prefix of variables names.
+            layer: Layer to add the variables to. The variables are saved
+            in the layer's scope.
+
+        Returns:
+            Dictionary of new variables.
+        """
+        ptq_threshold_tensor = layer.add_weight(
+            name + THRESHOLD_TENSOR,
+            shape=(),
+            initializer=tf.keras.initializers.Constant(1.0),
+            trainable=False)
+        ptq_threshold_tensor.assign(self.np_threshold_values)
+
+        fq_min = layer.add_weight(
+            name + FQ_MIN,
+            shape=(),
+            initializer=tf.keras.initializers.Constant(-1.0),
+            trainable=False)
+        fq_min.assign(self.min)
+
+        fq_max = layer.add_weight(
+            name + FQ_MAX,
+            shape=(),
+            initializer=tf.keras.initializers.Constant(1.0),
+            trainable=False)
+        fq_max.assign(self.max)
+
+        # save the quantizer added parameters for later calculations
+        self.quantizer_parameters = {THRESHOLD_TENSOR: ptq_threshold_tensor,
+                                     FQ_MIN: fq_min, FQ_MAX: fq_max}
+        return self.quantizer_parameters
+
+    def __call__(self,
+                 inputs: tf.Tensor,
+                 training: bool):
+        """
+        Quantize a tensor.
+        Args:
+            inputs: Input tensor to quantize.
+            training: Whether the graph is in training mode.
+
+        Returns:
+            The quantized tensor.
+        """
+
+        _min = self.quantizer_parameters[FQ_MIN]
+        _max = self.quantizer_parameters[FQ_MAX]
+        q_tensor = tf.quantization.fake_quant_with_min_max_vars(inputs, _min, _max,
```

Comment:
Where are the constraints for symmetric or PO2 ? Am I missing something?

Reply:
will be added next PR